src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs

```diff
@@ -2606,5 +2606,51 @@ public S(object x, object y)
                 //         void f() { F = x; }
                 Diagnostic(ErrorCode.ERR_ThisStructNotInAnonMeth, "F").WithLocation(7, 20));
         }
+
+        [Fact]
+        public void TestUnusedInterpolatedString()
+        {
+            var program = @"
+            public class Program
+            {
+                public static void Main(string[] args)
+                {
+                    var constString = ""abc"";
+                    var constInterpolatedString1 = $""abc"";
+                    var constInterpolatedString2 = $""ab{1}c"";
+                    var constInterpolatedString3 = $""ab{""1""}c"";
+                    var constNestedInterpolatedString = $""{$""a""}abc"";
+
+                    var nonConstInterpolatedString1 = $""Function: {GetFunction()}"";
+                    var nonConstInterpolatedString2 = $""Function: {GetProperty}"";
+
+                    var variable = ""abc"";
+                    var nonConstInterpolatedString3 = $""Variable: {variable}"";
+                }
+
+                public static string GetFunction() => ""function"";
+                public static string GetProperty => ""property"";
+            }";
+ 
+            var comp = CreateCompilation(program);
+            comp.VerifyDiagnostics
+            (
+                // (6,25): warning CS0219: The variable 'constString' is assigned but its value is never used
+                //                     var consString = "abc";
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "constString").WithArguments("constString").WithLocation(6, 25),
+                // (7,25): warning CS0219: The variable 'constInterpolatedString1' is assigned but its value is never used
```

Comment:
> (7,25): warning CS0219: The variable 'constInterpolatedString1' is assigned but its value is never used [](start = 18, length = 104)

This new warning is a compat break. We can run it by the compat council, but I don't think it  would be accepted (annoyance potential versus value trade-off seems low). So I think we should wait  for warning waves infrastructure, and then fix this issue in one of the waves.